packages/remark-lint-no-undefined-references/index.js

```diff
@@ -123,9 +137,7 @@ const remarkLintNoUndefinedReferences = lintRule(
     const contents = String(file)
     const loc = location(file)
     const lineEnding = /(\r?\n|\r)[\t ]*(>[\t ]*)*/g
-    const allow = new Set(
-      (option.allow || []).map((d) => normalizeIdentifier(d))
-    )
+    const allow = new AllowSet(option.allow || [])
```

Comment:
I don’t think a whole custom class is needed here. We very rarely use classes.

Reply:
E.g., how about this:

```suggestion
/** @type {Array<RegExp>} */
const regexes = [];
/** @type {Array<string>} */
const strings = []

let index = -1;

while (++index < allow.length) {
    const value = allow[index];
    if (typeof value === 'string') {
        strings.push(value)
    } else {
        regexes.push(new RegExp(value.source, 'i'))
    }
}
```

Reply:
The class creates an abstraction around checking whether an identifier matches one of the allow values which isn't simple a simple set membership check anymore. The check occurs in two places, so I thought it would be nice to hide the checking logic.

Would you rather prefer your suggestion plus a closure below your suggested snippet that abstracts the check?

Reply:
Yes, I understand the need to abstract the check itself. Indeed, I’d prefer a closure below!

Reply:
See 8c27df2dc6820b9a6bb67eba7f3dc1a49f9ab6da. WDYT?